### PR TITLE
Datacatalog delete artifact

### PR DESCRIPTION
**Pull request by MorpheusXAUT**
# TL;DR
This PR prepares datacatalog and flyteadmin endpoints for the second part of our cache eviction changes, evicting the cache of a (past) execution.

## Type
 - [ ] Bug Fix
 - [X] Feature
 - [ ] Plugin

## Are all requirements met?

 - [X] Code completed
 - [X] Smoke tested
 - [ ] Unit tests added
 - [X] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
A new `CacheService` was introduced to flyteadmin's proto definitions. The two currently defined methods allow for cache eviction for an `Execution` or `TaskExecution`. Initially, both endpoints were merged in a single endpoint (since handling in flyteadmin is very similar), however due to an [issue](https://github.com/grpc-ecosystem/grpc-gateway/issues/753) in `grpc-gateway < 2.0.1` with `oneof` properties and `additional_bindings`, I've had to split them into two separate endpoints. This could be be merged again once the dependency upgrade has completed, however too many breaking changes would be introduced at the moment.
The returned `CacheEvictionErrors` are intended to be displayed in the frontend to provide feedback about cache eviction to the user.

Additionally, datacatalog's API was extended by `DeleteArtifact`, allowing for an artifact to be deleted from the database/blob storage based on ID or tag.

## Tracking Issue
https://github.com/flyteorg/flyte/issues/2867

## Follow-up issue
_NA_

### CODE REVIEW

**Comment by MorpheusXAUT** [comment]
Rebased onto current `master` branch.
